src/SDDK/wf_inner.cpp

```diff
@@ -35,61 +35,77 @@
 
 namespace sddk {
 
-namespace {
+// define type traits for a single template implementation of both real and complex matrix
+// general case for real matrix
+template <typename T>
+struct real_type {using type = T;};
+
+// special case for complex matrix
 template <typename T>
-void scale_gamma_wf(spin_range spins, int m, int i0, double alpha, Wave_functions& bra);
+struct real_type<std::complex<T>> {using type = T;};
+
+namespace {
+//template <typename T>
+//void scale_gamma_wf(spin_range spins, int m, int i0, double alpha, Wave_functions& bra);
 
 // If scalar type, g-vector 0 contribution must be scaled before / after inner product to aboid counting twice
-template <>
-void scale_gamma_wf<double>(spin_range spins, int m, int i0, double alpha, Wave_functions& bra) {
+template <typename F, typename T>
```

Comment:
Out of my own curiosity: why there are two types: F and T?

Reply:
It was originally an error to compile that without this definition but now it is fixed using real_type